backend/YouTubeV2.Application/Services/UserService.cs

```diff
@@ -0,0 +1,32 @@
+﻿using Microsoft.AspNetCore.Identity;
+using YouTubeV2.Application.DTO;
+using YouTubeV2.Application.Exceptions;
+using YouTubeV2.Application.Model;
+
+namespace YouTubeV2.Application.Services
+{
+    public class UserService
+    {
+        private readonly YTContext _context;
+        private readonly UserManager<User> _userManager;
+
+        public UserService(YTContext context, UserManager<User> userManager)
+        {
+            _context = context;
+            _userManager = userManager;
+        }
+
+        public async Task RegisterAsync(RegisterDto registerDto, CancellationToken cancellationToken)
+        {
+            var user = new User(registerDto);
+
+            var result = await _userManager.CreateAsync(user, registerDto.password);
+            if (!result.Succeeded)
+                throw new BadRequestException(result.Errors.Select(error => new ErrorResponseDTO(error.Description)));
```

Comment:
Idk if thats a good exception name. I know where you get that name from but idk if it is giving us the idea of what actually failed

Reply:
also idk if having exception that is trying to represent multiple things at once is a good idea maybe just pick the first error

Reply:
I think the mesage gives a good idea of what failed.
If the erorrs are something like "password must be longer than 8 characters" and "password must have cappital letter" i think its good to have both.